build.assets/Dockerfile-arm

```
@@ -1,8 +1,72 @@
-ARG BUILDBOX_VERSION
-FROM public.ecr.aws/gravitational/teleport-buildbox:$BUILDBOX_VERSION
+FROM docker.io/library/debian:11
 
-USER root
+COPY locale.gen /etc/locale.gen
+COPY profile /etc/profile
+
+ENV LANGUAGE="en_US.UTF-8" \
+    LANG="en_US.UTF-8" \
+    LC_ALL="en_US.UTF-8" \
+    LC_CTYPE="en_US.UTF-8" \
+    DEBIAN_FRONTEND="noninteractive"
+
+# BUILDARCH is automatically set by DOCKER when building the image with Build Kit (MacOS by default).
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+ARG BUILDARCH
 
 RUN apt-get -y update && \
-    apt-get -y install gcc-arm-linux-gnueabihf gcc-aarch64-linux-gnu && \
-    apt-get -y autoclean && apt-get -y clean
+    apt-get -y install software-properties-common && \
+    apt-get update -y --fix-missing && \
+    apt-get -q -y upgrade && \
+    apt-get install -q -y --no-install-recommends \
+        apt-utils \
+        build-essential \
+        ca-certificates \
+        curl \
+        git \
+        gnupg \
+        gzip \
+        libc6-dev \
+        libpam-dev \
+        locales \
+        pkg-config \
+        sudo \
+        unzip \
+        zip \
+        # ARM dependencies
+        libc6-dev-armhf-cross \
+        gcc-arm-linux-gnueabihf \
+        && \
+    dpkg-reconfigure locales && \
+    apt-get -y clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Node.js.
+ARG NODE_VERSION
+ENV NODE_PATH="/usr/local/lib/nodejs-linux"
+ENV PATH="$PATH:${NODE_PATH}/bin"
+RUN export NODE_ARCH=$(if [ "$BUILDARCH" = "amd64" ]; then echo "x64"; else echo "arm64"; fi) && \
+    export NODE_URL="https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-${NODE_ARCH}.tar.xz" && \
+    mkdir -p ${NODE_PATH} && \
+    curl -o /tmp/nodejs.tar.xz -fsSL ${NODE_URL} && \
+    tar -xJf /tmp/nodejs.tar.xz -C /usr/local/lib/nodejs-linux --strip-components=1
```

nit: Bash style, proper quoting, exports unnecessary.

```suggestion
RUN NODE_ARCH="$(if [ '$BUILDARCH' = 'amd64' ]; then echo 'x64'; else echo 'arm64'; fi)" && \
    NODE_URL="https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-$NODE_ARCH.tar.xz" && \
    mkdir -p "$NODE_PATH" && \
    curl -o /tmp/nodejs.tar.xz -fsSL "$NODE_URL" && \
    tar -xJf /tmp/nodejs.tar.xz -C /usr/local/lib/nodejs-linux --strip-components=1
```

---

"proper quoting" around `"$BUILDARCH"` breaks it as you want to evaluate the value and not take the literal. Applied everything else.